tests/unit/test_charm.py

```diff
@@ -631,3 +629,98 @@ def test_given_self_signed_option_is_false_and_unit_is_leader_when_generate_cert
         event.fail.assert_called_once_with(
             message="Action not supported as charm is not configured to be self-signed"
         )
+
+    def test_given_no_requirer_application_when_get_all_certificate_requests_action_then_empty_list_returned(  # noqa: E501
+        self,
+    ):
+        event = Mock()
+        self.harness.charm._on_get_all_certificate_requests_action(event=event)
+        event.set_results.assert_called_once_with({"Result": []})
+
+    @patch(
+        "charms.tls_certificates_interface.v2.tls_certificates.TLSCertificatesProvidesV2.get_requirer_units_csrs_with_no_certs"  # noqa: E501, W505
+    )
+    def test_given_requirer_application_when_get_all_certificate_requests_action_then_csrs_information_is_returned(  # noqa: E501
```

Comment:
Do we need another test to validate that if we provide a certificate, the request does not show up anymore?

Reply:
That is validated in the library, do you think here is a good place to validate it too?
This `test_given_csrs_with_certs_issued_when_get_requirer_csrs_with_no_certs_then_the_info_of_those_certs_not_returned` test case in [this PR](https://github.com/canonical/tls-certificates-interface/pull/62).